test cases/common/198 dict/meson.build

```diff
@@ -0,0 +1,12 @@
+project('dict test', 'c')
+
+dict = {foo : 'bar',
+        baz : 'foo',
+        foobar: 'baz'}
```

Comment:
Also it might make sense to have assignment operator similar to `+=` for arrays. It would create a copy of the dict with additional/overridden value.

Reply:
this hinges on the decision of whether a dict should be a mutable object or not, I didn't want to make arguing my case more complicated than strictly needed tbh ;)

Reply:
Sure, we could introduce it later if needed, let's keep it simple for now. It would not make dicts mutable though as semantics would be same as in arrays.